### DISCUSSION TITLE
Random disk assignation for NVME disk on baremetal

### DISCUSSION THREAD

**Discussion by piellick** [siderolabs/talos#10049] [Q&A] [answered]
Hi everyone,

we are facing a strange behavior on disk assigniation. It was tested with talOS `1.8.3` and `1.9` .

We configure talos with :
```
machine:
  install:
    disk: /dev/nvme0n1
```

We used baremetal with two NVME:
- `nvme0n1` 
- `nvme1n1` 

On maintenance mode, `talosclt get disks` return correct disk list with the right path (`/dev/nvme0n1` and `/dev/nvme1n1`).
I could provide some captures later.

Talos Deployment work well but assign randomly nvme0 or nvme1 for OS deployment. Someone already had this case?

thanks a lot :pray:

**Comment by smira** [comment, marked as answer]
This is not an issue in Talos, but rather the way hardware and Linux works. Disk enumeration is random, so whatever was `nvme0` might on next boot become `nvme1`.

You can use either device selectors, or `/dev/by-id` symlinks to provide a stable assignment (if required).

It's not a problem for Talos install, as this instruction is used only once on initial install, and later on Talos will find its install disk, no matter what name it has.

**Comment by piellick** [comment]
hi @smira this confirms our investigation. Thanks a lot !
